assets/js/base/components/payment-methods/payment-methods.js

```diff
@@ -0,0 +1,94 @@
+/**
+ * External dependencies
+ */
+import {
+	useCheckoutData,
+	usePaymentEvents,
+	useActivePaymentMethod,
+} from '@woocommerce/base-hooks';
+import { getPaymentMethods } from '@woocommerce/blocks-registry';
+import { useCallback } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+
+/**
+ * Internal dependencies
+ */
+import Tabs from '../tabs';
+
+const noPaymentMethodTab = () => {
+	const label = __( 'Not Existing', 'woo-gutenberg-products-block' );
+	return {
+		name: label,
+		label,
+		title: label,
+	};
+};
+
+const createTabs = ( paymentMethods ) => {
+	const paymentMethodsKeys = Object.keys( paymentMethods );
+	return paymentMethodsKeys.length > 0
+		? paymentMethodsKeys.map( ( key ) => {
+				const { tab, ariaLabel } = paymentMethods[ key ];
+				return {
+					name: key,
+					title: tab,
+					ariaLabel,
+				};
+		  } )
+		: [ noPaymentMethodTab() ];
+};
+
+const paymentMethods = getPaymentMethods();
```

Comment:
Hmm good point.

- for payment methods that depend on specific apis, I would expect the onus is on them to only register if those api's are present.
- perhaps the payment method should have a way to remove itself from being registered if later during initialization process it determines it doesn't have the apis it needs?
- otherwise, I think the logic involved here is intentionally passed off to the payment method and it could decide how it presents any related information to what it is able to do or not given the data it receives?

Reply:
> perhaps the payment method should have a way to remove itself from being registered if later during initialization process it determines it doesn't have the apis it needs?

That should work for things like Apple Pay (only available in Safari) or Payment Request (also browser-dependent). Or even better, the gateway initialization logic can just *not* register the button if it detects that the browser is not compatible.

I can't think about ay use-case more complex than that. There are some country-specific gateways, but they depend on the merchant's country, not the customers'. If y'all can point to a gateway that should be shown/hidden depending on the `country` field on the checkout, then we'll need a more sophisticated solution.

Reply:
> perhaps the payment method should have a way to remove itself from being registered if later during initialization process

Yeah, I like that idea. One use case which I've seen is when the Apple Pay API exists in the browser (ie: we are in Safari), but `canMakePayments` (an async operation) returns false for some reason (eg: no https) during initialization.

Reply:
implemented in d6c7e4f6d0b21c0a4